src/passes/MemoryPacking.cpp

```diff
@@ -27,48 +28,78 @@ struct MemoryPacking : public Pass {
   bool modifiesBinaryenIR() override { return false; }
 
   void run(PassRunner* runner, Module* module) override {
-    if (!module->memory.exists) return;
+    if (!module->memory.exists || runner->options.features.hasBulkMemory()) {
```

Comment:
please explain why in a comment. I actually don't know the answer myself...?

Reply:
Wil do. We conservatively do not change any segments to avoid invalidating segment indices or contents that might be referenced from a memory.init instruction.